docs/docsite/rst/dev_guide/developing_python3.rst

```diff
@@ -274,67 +374,42 @@ coded to expect bytes on Python-2 and text on Python-3.
 Tips, tricks, and idioms to adopt
 =================================
 
-Exceptions
-----------
+Old Exception Syntax
+--------------------
 
-In order for code to function on Python-2.6+ and Python-3, use the
-new exception-catching syntax which uses the ``as`` keyword::
+Until Ansible-2.4, modules needed to be compatible with Python-2.4 as
+well.  Python-2.4 did not understand the new exception-catching syntax so
+we had to write a compatibility function that could work with both
+Python-2 and Python-3.  You may still see this used in some modules:
+
+.. code-block:: python
+
+    from ansible.module_utils.pycompat24 import get_exception
+    [...]
 
     try:
         a = 2/0
-    except ValueError as e:
```

Comment:
(A) is what is there now.  Added (B).    (C) is documented in the module section. (D) is a global thing and we haven't really nailed down a decision here.  I skirt around the issue i nthe module section with this wording:

"Unless a change is going to be backported to Ansible-2.3, you should not
have to use this in new code." (where "this" refers to the get_exception() based coding style).